include/libnuraft/raft_server.hxx

```diff
@@ -743,9 +743,16 @@ public:
      * Manually create a snapshot based on the latest committed
      * log index of the state machine.
      *
-     * @return `true` on success.
+     * @return `true` on success or on scheduled for asynchronously creation.
+     */
+    ulong create_snapshot();
+
+    /**
+     * Check whether a snapshot is created
+     * @param log_idx
+     * @return true if last_log_idx_ of last snapshot or equal with or bigger than log_idx
      */
-    bool create_snapshot();
+    bool snapshot_created(ulong log_idx);
```

Comment:
This API is too ad-hoc. I'd rather add an API returning the log index of the last snapshot:
```c++
/**
 * Get the log index number of the last snapshot.
 *
 * @return Log index number of the last snapshot.
 *         `0` if snapshot does not exist.
 */
ulong get_last_snapshot_idx() const;
```